### DISCUSSION TITLE
Get all folders from folder

### DISCUSSION THREAD

**Discussion by pINKmUNster** [pnp/pnpcore#378] [Q&A]
Hello, 

Does the SDK provide a method to get all folder hierarchy from a specified folder?
Right now I am making this recursively but it's take a lot time and make a lot of call to SharePoint.

Thanks.

**Comment by jansenbe** [comment]
There's no easy way doing this non-recursively as there are no SharePoint folder API's that do the traversal server side.  One option to play with is doing a CAML based retrieval of the folders (https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html). You can filter on folders using below CAML snippet (not tested by me):

```xml
<View Scope='RecursiveAll'><Query><Where><Eq><FieldRef Name='FSObjType'/><Value Type='Integer'>1</Value></Eq></Where></Query></View>
```

**Comment by pINKmUNster** [comment]
Ok, thank a lot. 
I will try with CAML.

**Reply by jansenbe** [reply]
@pINKmUNster : if you've fixed this then would be good if you could paste the resulting CAML here and mark this as answered. Will help others facing the same question :-)
